src/classes/THAN_ClearCache_TEST.cls

```diff
@@ -0,0 +1,69 @@
+/*
+    Copyright (c) 2017, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group TDTM
+* @group-content ../../ApexDocContent/TDTM.htm
+* @description Test class for the THAN_ClearCache_TDTM class
+*/
+@isTest
+public with sharing class THAN_ClearCache_TEST {
+
+    //Test to verify that the cached tokens get cleared
+    @isTest
+    public static void clearCacheAfterInsertTriggerHandler() {
+        //Insert user's handlers
+        Trigger_Handler__c contactUpdate = new Trigger_Handler__c(
+                Active__c = true,
+                Asynchronous__c = false,
+                Class__c = 'Contact_Update_User',
+                Load_Order__c = 0.20,
+                Object__c = 'Contact',
+                Trigger_Action__c = 'BeforeInsert;BeforeUpdate',
+                Filter_Field__c = 'Gender__c',
+                Filter_Value__c = 'Male'
+        );
+
+        insert contactUpdate;
+
+        //After insert Trigger Handler, system should clear the cache, so getTdtmConfig should include the new TriggerHandler
+        List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getTdtmConfig();
```

Comment:
Is `TDTM_Global_API.getTdtmConfig()` going to return only the TDTM record we manually inserted, or the default config plus the one we inserted? Can we assert the size of the returned list?

Reply:
Here we should only get the TDTM record that we manually created. Since we didn't insert the default config using setTdtmConfig. I will put the assertion here to verify that the size is 1

Reply:
Here we should only get the TDTM record that we manually created. Since we didn't insert the default config using setTdtmConfig. I will put the assertion here to verify that the size is 1